core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -319,8 +320,8 @@ private DefaultFunctionResolver dayOfWeek() {
    * DAYOFYEAR(STRING/DATE/DATETIME/TIMESTAMP).
    * return the day of the year for date (1-366).
    */
-  private DefaultFunctionResolver dayOfYear() {
-    return define(BuiltinFunctionName.DAYOFYEAR.getName(),
+  private DefaultFunctionResolver dayOfYear(BuiltinFunctionName dayOfYear) {
+    return define(dayOfYear.getName(),
```

Comment:
Same as https://github.com/Bit-Quill/opensearch-project-sql/pull/175#discussion_r1031842704
Function should accept `TIME` argument.

Reply:
```
mysql> select dayofyear(curtime());
+----------------------+
| dayofyear(curtime()) |
+----------------------+
|                  329 |
+----------------------+
1 row in set (0.00 sec)
```

Reply:
Will be accomplished in future work
https://bitquill.atlassian.net/browse/AOS-1266